libminifi/src/utils/OsUtils.cpp

```diff
@@ -78,6 +81,8 @@ std::string OsUtils::userIdToUsername(const std::string &uid) {
     // First call to LookupAccountSid to get the buffer sizes.
     PSID pSidOwner = NULL;
     if (ConvertStringSidToSidA(name.c_str(), &pSidOwner)) {
+      const utils::ScopeGuard guard_pSidOwner([pSidOwner]() { LocalFree(pSidOwner); });
```

Comment:
MSDN doesn't seem to guarantee that `LocalFree` only has to be called when the function call is successful. I suggest moving the scope guard outside with a null check.

Reply:
Done